examples/bri-3/src/bri/identity/bpiSubjectAccounts/agents/bpiSubjectAccountsStorage.agent.ts

```diff
@@ -0,0 +1,137 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from '../../../../../prisma/prisma.service';
+import { BpiSubject } from '../../bpiSubjects/models/bpiSubject';
+import { NOT_FOUND_ERR_MESSAGE } from '../api/err.messages';
+import { BpiSubjectAccount } from '../models/bpiSubjectAccount';
+
+// Repositories are the only places that talk the Prisma language of models.
+// They are always mapped to and from domain objects so that the business layer of the application
+// does not have to care about the ORM.
+@Injectable()
+export class BpiSubjectAccountStorageAgent extends PrismaService {
+  async getBpiSubjectAccountById(id: string): Promise<BpiSubjectAccount> {
+    const bpiSubjectAccountModel = await this.bpiSubjectAccount.findUnique({
+      where: { id: id },
+      include: { OwnerBpiSubject: true, CreatorBpiSubject: true },
+    });
+
+    if (!bpiSubjectAccountModel) {
+      throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
+    }
+
+    const creatorBpiSubject = new BpiSubject(
```

Comment:
Current mapper can be used here in this file

Reply:
@Manik-Jain i think it would be better to leave todo until we decide which mapper to use? if we use automapper we will have to change this too

Reply:
Yeah, let's not duplicate work if new mapper is coming.